docs/remote/devcontainerjson-reference.md

```diff
@@ -59,6 +59,14 @@ While some devcontainer.json properties apply generally, others are only used in
 
 If you've already built the container and connected to it, be sure to run **Remote-Containers: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up the change.
 
+> **Note:** The format of certain properties will vary depending on the involvement of a shell.
+
+> `postCreateCommand`, `postStartCommand`, `postAttachCommand`, and `initializeCommand` all have an array and a string type, while `runArgs` only has the array variant. An array is passed to the OS for execution without going through a shell, whereas a string goes through a shell (it needs to be parsed into command and arguments).
```

Comment:
"only has the array type".
I was confused by the word "variant".

Reply:
Thanks for the feedback- updated.